velox/common/base/BitUtil.cpp

```diff
@@ -99,6 +99,15 @@ std::string toString(const uint64_t* bits, int offset, int size) {
   return ans;
 }
 
+void toString(const uint64_t* bits, int offset, int size, char* out) {
+  std::string ans;
```

Comment:
Do not allocate a new string here, just directly write to `out`, that's the whole point of this new function:

```c++
  for (int i = 0; i < size; ++i) {
    out[i] = '0' + isBitSet(bits, offset + i);
  }
```

We can also reuse this function in `std::string toString(const uint64_t* bits, int offset, int size)`

Reply:
done